src/keys/sodiumoxide.rs

```diff
@@ -0,0 +1,410 @@
+use crate::{
+    AsymmetricKeyBuilder, Buildable, Builder, ByteSealable, ByteUnsealable, BytesSources,
+    CryptoError, IntoIndex, KeyBuilder, PublicAsymmetricKeyBuilder, Sealable,
+    SecretAsymmetricKeyBuilder, States, Storer, SymmetricKeyBuilder, TypeBuilder,
+    TypeBuilderContainer, Unsealable, VectorBytesSource,
+};
+use async_trait::async_trait;
+use mongodb::bson::{self, Document};
+use serde::{Deserialize, Serialize};
+use sodiumoxide::crypto::{
+    box_::{
+        self,
+        curve25519xsalsa20poly1305::{
+            Nonce as ExternalSodiumOxideAsymmetricNonce,
+            PublicKey as ExternalSodiumOxidePublicAsymmetricKey,
+            SecretKey as ExternalSodiumOxideSecretAsymmetricKey,
+            PUBLICKEYBYTES as EXTERNALSODIUMOXIDEPUBLICASYMMETRICKEYBYTES,
+            SECRETKEYBYTES as EXTERNALSODIUMOXIDESECRETASYMMETRICKEYBYTES,
+        },
+    },
+    secretbox::{
+        self,
+        xsalsa20poly1305::{
+            Key as ExternalSodiumOxideSymmetricKey, Nonce as ExternalSodiumOxideSymmetricNonce,
+            KEYBYTES as EXTERNALSODIUMOXIDESYMMETRICKEYBYTES,
+        },
+    },
+};
+use std::{boxed::Box, convert::TryFrom};
+
+// SYMMETRIC KEY \\
+#[derive(Serialize, Deserialize, Debug, Clone)]
+pub struct SodiumOxideSymmetricKeySealable {
+    pub source: BytesSources,
+    pub key: Box<States>,
+    pub nonce: ExternalSodiumOxideSymmetricNonce,
+}
+
+#[async_trait]
+impl Sealable for SodiumOxideSymmetricKeySealable {
+    async fn seal<T: Storer>(self, storer: T) -> Result<ByteUnsealable, CryptoError> {
+        let stateful_key = *self.key.clone();
```

Comment:
should `self.key` be renamed `self.stateful_key`?  Because in the next line there is a new `key` variable which is not the same thing, confusing.